planner/core/rule_aggregation_push_down.go

```diff
@@ -427,7 +424,24 @@ func (a *aggregationPushDownSolver) aggPushDown(p LogicalPlan) (_ LogicalPlan, e
 				}
 				projChild := proj.children[0]
 				agg.SetChildren(projChild)
-			} else if union, ok1 := child.(*LogicalUnionAll); ok1 {
+			} else if union, ok1 := child.(*LogicalUnionAll); ok1 && p.SCtx().GetSessionVars().AllowAggPushDown {
+				for _, aggFunc := range agg.AggFuncs {
```

Comment:
What's the difference between `line 428~443` and `line 445~460`?

Reply:
There's no difference.